### PR TITLE
fix(argo-frontend-plugin): skip helm revision metadata

### PR DESCRIPTION
**Pull request by janhoon**
In cases where argo application history contains revisions that were deployed via helm, the api calls to `/api/v1/applications/{name}/revisions/{revision}/metadata` would return a 500. This is due to the call requiring the revision param to be a git sha, but in the case of helm it will be a version (and there would be no repo to read from.)

ArgoCD source for the above here: https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L2154-L2156

#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [x] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by janhoon** [comment]
This is a fix for https://github.com/RoadieHQ/roadie-backstage-plugins/issues/980, just need some tests for the changes

File: plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts [hunk 1]
```diff
@@ -41,6 +41,9 @@ export const useAppDetails = ({
   ) => {
     const promises: Promise<void>[] | undefined =
       appDetails.status?.history?.map(async (historyRecord: any) => {
+        if (historyRecord.source?.chart !== undefined) {
```
Comment (janhoon) [inline plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts:44]: From the argo swagger docs for getting an application (`/api/v1/applications/{name}`) for `status.history[].source.helm: 

> Chart is a Helm chart name, and must be specified for applications sourced from a Helm repo.

**Comment by janhoon** [comment]
The result of this PR is ultimately that 2 columns in ArgoCD history is blank if the source is helm:

image

Could add a unit test for this, but don't feel it would be right to check if fields are empty if the source is helm, happy to do it, but will wait for a review

File: plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts (hunk 1 above)
Comment (janhoon) [reply plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts:45]: if this is defined revision will be a helm chart version and not git sha

**Review by Xantier — APPROVED** [review]
